### PR TITLE
PUBDEV-6694: add AUCPR as early stopping metric

### PR DESCRIPTION
**Pull request by wendycwong**
This PR completes the work for JIRA: https://0xdata.atlassian.net/projects/PUBDEV/issues/PUBDEV-6684?filter=myopenissues&orderby=priority%20DESC

Work done:
1. Added java backend support to allow AUCPR as an early-stopping parameter.
2. Added Runit and Pyunit test to ensure correct operation.

@ledell @NidhiMehta 
Please let me know if AUCPR is an okay name for the parameter.  I can change it to anything you guys want.  Just let me know please.

### CODE REVIEW

File: h2o-core/src/main/java/hex/ScoreKeeper.java [hunk 1]
```diff
@@ -126,9 +126,10 @@ else if (m instanceof ModelMetricsMultinomial) {
     RMSE(ConvergenceStrategy.LESS_IS_BETTER, true),
     MAE(ConvergenceStrategy.LESS_IS_BETTER, true),
     RMSLE(ConvergenceStrategy.LESS_IS_BETTER, true),
-    AUC(ConvergenceStrategy.MORE_IS_BETTER, false),
```
Comment (michalkurka) [inline h2o-core/src/main/java/hex/ScoreKeeper.java]: the lower bound only matters to LESS_IS_BETTER, imho it doesn't matter whether we have true or false there - but obviously AUC is actually lower bound by 0

File: h2o-core/src/test/java/water/TestUtil.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.rules.TestRule;
+import org.junit.runner.Computer;
```
Comment (michalkurka) [inline h2o-core/src/test/java/water/TestUtil.java]: unused import?

**Comment by michalkurka** [comment]
@wendycwong the commit history includes other people's commits

can you please rebase this on top of rel-yau and set the target branch to rel-yau?

File: h2o-core/src/main/java/hex/ScoreKeeper.java (hunk 1 above)
Comment (wendycwong) [reply h2o-core/src/main/java/hex/ScoreKeeper.java]: Got it.

Comment (wendycwong) [reply h2o-core/src/main/java/hex/ScoreKeeper.java]: Can I keep it as true though?  It looks better.

Comment (michalkurka) [reply h2o-core/src/main/java/hex/ScoreKeeper.java]: yes, no harm there - I agree it does look better

**Pull request merged**